character-sfx.js

```diff
@@ -342,8 +351,49 @@ class CharacterSfx {
   
     };
     _handleNarutoRun();
-    
 
+    // combo
+    const dispatchComboSoundEvent = (soundIndex) =>{
+      this.dispatchEvent(new MessageEvent('meleewhoosh', {
+        data: {
+          index: soundIndex
+        },
+      }));
+    }
+    const _handleCombo = () => {
+      if (this.player.hasAction('use') && this.player.getAction('use').behavior === 'sword' && this.player.getAction('use').animationCombo) {
+        const comboAnimationName = aimAnimations[this.player.getAction('use').animationCombo[this.player.avatar.useAnimationIndex]];
+        if (comboAnimationName && comboAnimationName !== this.lastSwordComboName) {
+          this.swordComboStartTime = timeSeconds;
+          this.alreadyPlayComboSound = false;
+        }
+        if (comboAnimationName) {
+          const animations = Avatar.getAnimations();
+          const animation = animations.find(a => a.name === comboAnimationName);
+          const animationComboIndices = Avatar.getanimationComboIndices();
+          const animationIndices = animationComboIndices.find(i => i.name === comboAnimationName);
+          const handDeltas = this.player.getAction('use').boneAttachment === 'leftHand' ? animationIndices.rightHandDeltas : animationIndices.leftHandDeltas;
+          const maxDeltaIndex = this.player.getAction('use').boneAttachment === 'leftHand' ? animationIndices.maxRightDeltaIndex : animationIndices.maxLeftDeltaIndex;
+          
+          const ratio = (timeSeconds - this.swordComboStartTime) / animation.duration;
+          if (ratio <= 1 && !this.alreadyPlayComboSound) {
+            const index = Math.floor(ratio * handDeltas.length);
+            if (index > maxDeltaIndex) {
+              this.alreadyPlayComboSound = true;
+              this.playGrunt('attack');
+              const soundIndex = this.player.avatar.useAnimationIndex * 4 + Math.floor(Math.random() * 4);

```

Comment:
Why `4`? Don't include random constants in the middle of the code.

Reply:
got it
[remove constsant in combo sound selection](https://github.com/webaverse/app/pull/3674/commits/9af4f0458f09df5eeb9d736a41cba5c5a50b9956)